closure.ts

```diff
@@ -0,0 +1,348 @@
+import { Stmt, Expr, Type, UniOp, BinOp, Literal, Program, FunDef, VarInit, Class, SourceLocation } from './ast';
+import { TypeCheckError } from './error_reporting'
+import { NUM, BOOL, NONE, CLASS } from './utils';
+
+let closureClassCounter = 0
+
+export function translateClosuresToClasses(program : Program<SourceLocation>): Program<SourceLocation> {
+  closureClassCounter = 0
+
+  program.funs.forEach(func => translateClosuresInFunc(func, program))
+  // TODO methods in class might also contain closures
```

Comment:
Is this done?

Reply:
This is not done. Since our code currently only loops through top level functions and parse the closures inside. Closures in methods of a class are currently ignored. We will add this functionality later.